song-server/src/main/java/bio/overture/song/server/model/analysis/Analysis.java

```diff
@@ -100,4 +114,23 @@ public AnalysisTypeId getAnalysisType() {
   public void setAnalysisState(String state) {
     this.analysisState = resolveAnalysisState(state).toString();
   }
+
+  public void populatePublishTimes() {
+    val history = this.analysisStateHistory;
+
+    if (!history.isEmpty()) {
+      val publishHistory =
+          history.stream()
+              .filter(stateChange -> stateChange.getUpdatedState().equals(PUBLISHED.name()))
+              .map(stateChange -> stateChange.getUpdatedAt())
+              .collect(toImmutableList());
```

Comment:
replace this with
```java
              .collect(toUnmodifiableList());


```
back before java11 we depended on `toImmutableList()` from an external library, but since then java11 add it in.